src/gui/render.cpp

```diff
@@ -412,7 +412,18 @@ static void RENDER_Reset( void ) {
 			gfx_flags = (gfx_flags & ~GFX_CAN_8) | GFX_RGBONLY;
 			break;
 	}
-	gfx_flags=GFX_GetBestMode(gfx_flags);
+	gfx_flags=GFX_GetBestMode(gfx_flags); 
+	if(
+		gfx_flags & GFX_UNITY_SCALE &&
+		complexBlock == 0 &&
+		strstr( simpleBlock->name, "Normal" ) == simpleBlock->name) { 
+	// Ant_222 simply setting GFX_SCALING doesn't work
+	// Ant_222: bad idea to set UNITY_SCALE here, because later, in SetSize(),
+	//          we may have to revert to the simple surface mode, but it will be late.
+		gfx_scalew = 1.0; gfx_scaleh = 1.0;
+		xscale     = 1  ; yscale     = 1;
+		simpleBlock = &ScaleNormal1x;
+	}
```

Comment:
1, 2, and 4 fixed, although one statement per line is too cumbersome for those simple and analogous (width and height, x and y) assignments.

I have removed the comment about `GFX_SCALING`, but the other one is up to you. My opinion is that `goto`-based fall-back logic in this spaghetti code is too difficult to test and to keep track of. The problem is that we may have to forego `GFX_UNITY_SCALE` if *pixel-perfect* could not be activated. If the fall-back logic does include another invocation of `RENDER_Reset`, we shall be stuck with `GFX_UNITY_SCALE`.

Reply:
There is not such thing as `simpleBlock->is_normal`. Why do you object to in-place string comparison when this comparison is used only once? It is not strictly necessary to wrap it into a bool member...

Reply:
I object to string comparison, because it suggests this value can somehow be injected by user, when that's no the case - this is purely internal name hardcoded during compilation, so we can as well hardcode the boolean. The change to introduce `is_normal` is somewhat separate from pixel-perfect scaling code, so can be implemented as a separate change, it's not a blocker for this review.

Reply:
> Have you tested that `complexBlock == 0` implies `simpleBlock != 0`?

No, I did not. If you depend on complexBlock not being selected, then I suggest adding another variable to communicate your intent to the reader:

    const bool simple_block_used = (complexBlock == nullptr);

And then use is inside `if`.